app/controllers/provider_management/external_users_controller.rb

```diff
@@ -65,34 +66,23 @@ def update_password
     end
   end
 
-  def confirmation
-    if @external_user.active?
-      render :disable_confirmation
-    else
-      render :enable_confirmation
-    end
+  def change_availability
+    render @external_user.active? ? :disable_confirmation : :enable_confirmation
   end
 
-  def enable
-    if (@external_user.provider == @provider) && @external_user.softly_deleted? && @external_user.un_soft_delete
-      redirect_to_show_page(notice: I18n.t('provider_management.external_users.enable_confirmation.success_message'))
+  def update_availability
+    if external_user_params[:availability].eql?('true')
```

Comment:
You can use `ActiveRecord::Type::Boolean.new.cast` to test boolean params, which would make this;

```ruby
    ActiveRecord::Type::Boolean.new.cast(params[:availablability]) ? enable : disable
```

(I cannot make this a 'suggestion' as Github doesn't see every line as changed)

Reply:
have used `ActiveModel::Type::Boolean.new.cast` instead following suggestions on SO.